vllm_ascend/distributed/llmdatadist_connector_v1.py

```diff
@@ -0,0 +1,969 @@
+                                           candidate_keys, self._tp_size))
+
+    def _get_first_matching_value(self, config_dict: dict,
+                                  candidate_keys: List[str],
+                                  default: Any) -> Any:
+        for key in candidate_keys:
+            if key in config_dict:
+                return config_dict[key]
+        return default
+
+
+_CLUSTER_INFO: Optional["ClusterInfo"] = None
+
+
+def init_cluster_info(vllm_config: "VllmConfig") -> None:
+    global _CLUSTER_INFO
+    if _CLUSTER_INFO is not None:
+        raise ValueError("ClusterInfo is already initialized.")
+    _CLUSTER_INFO = ClusterInfo(vllm_config)
+
+
+def get_cluster_info() -> "ClusterInfo":
+    global _CLUSTER_INFO
+    if _CLUSTER_INFO is None:
+        raise ValueError("ClusterInfo is not initialized.")
+    return _CLUSTER_INFO
+
+
+def report_prefill_info(meta_server_url, prefill_info):
+    response = requests.post(f"{meta_server_url}/put", json=prefill_info)
+    if response.status_code != 200:
+        logger.error(
+            f"put_prefill_info failed status_code: {response.status_code}, response: {response.text}"
+        )
+
+
+def fetch_prefill_info(meta_server_url, request_ids):
+    response = requests.get(f"{meta_server_url}/get", json=request_ids)
+    if response.status_code != 200:
+        logger.error(
+            f"get_prefill_info failed status_code: {response.status_code}, response: {response.text}"
+        )
+        return None
+    return response.json()
+
+
+class KVTransferEngine:
+
+    def __init__(self, role: llm_datadist.LLMRole, local_rank: int,
+                 dp_rank: int, tp_rank: int, local_server_id: str) -> None:
+        self.role = role
+        self.local_rank = local_rank
+        self.tp_rank = tp_rank
+        self.cluster_info = get_cluster_info()
+
+        local_device_info = self.cluster_info.get_device(
+            local_server_id, dp_rank, tp_rank)
+        assert local_device_info is not None, \
+            "Could not find local device from cluster info."
+
+        self.cluster_id = local_device_info.cluster_id
+        self.local_device_ip = local_device_info.device_ip
+        self.datadist_engine = llm_datadist.LLMDataDist(
+            self.role, self.cluster_id)
+
+    def prepare_data_dist(self):
+        # TODO: The maximum size of the mbuf for the llm datadist. We need to
+        # find an appropriate value to minimize memory waste.
+        options = {
+            "llm.SyncKvCacheWaitTime": envs.LLMDATADIST_SYNC_CACHE_WAIT_TIME,
+            "ge.flowGraphMemMaxSize": f"{(9*1024*1024*1024):d}",
+            "ge.exec.deviceId": str(self.local_rank),
+        }
+        if self.role == llm_datadist.LLMRole.PROMPT:
+            options["llm.listenIpInfo"] = f"{self.local_device_ip}:26000"
+        self.datadist_engine.init(options)
+        self.kv_transfer = self.datadist_engine.kv_cache_manager
```

Comment:
AFAIK, we're working on replacing `kv_cache_manager` to `cache_manager`, @zzzzwwjj

Reply:
> AFAIK, we're working on replacing `kv_cache_manager` to `cache_manager`, @zzzzwwjj

@wangxiyuan  
Hi, I am playing around with cache_manager right now. But I find difficulties running the demo in https://gitee.com/ascend/samples/blob/master/python/level1_single_api/10_llm_data_dist/cache_manager_api_samples/pull_blocks_sample.py

with error below
```
llm_datadist.status.LLMException: [link] failed, error code is LLMStatusCode.LLM_LINK_FAILED, {1: 0, 2: 1}.
```

I am wondering if this this class is available in a single node case. In addition I added environment variable as the documentation stated here.
![image](https://github.com/user-attachments/assets/eda3dc43-e4af-4cf5-9d8c-1805aa25a5a1)